gradle-incremental-configuration-cache/src/test/groovy/com/palantir/gradle/configcache/incremental/IncrementalConfigurationCacheTest.groovy

```diff
@@ -89,7 +101,25 @@ class IncrementalConfigurationCacheTest extends IntegrationTestKitSpec {
         !output.contains('Configuration cache entry stored.')
     }
 
-    private boolean runTasksWithConfigurationCache(String... tasks) {
+    @Unroll
+    def "reassure users iff there are configuration cache problems"() {
+        file("gradle/configuration-cache-allowed-tasks") << ""
+
+        when:
+        def buildResult = createRunner([task, '--configuration-cache'] as String[]).build()
+
+        then:
+        buildResult.output.contains(
+                "[IncrementalConfigurationCachePlugin] ⚠️ Configuration Cache is being rolled out"
+        ) == shouldReassure
+
+        where:
+        task                         | shouldReassure
+        "breaksConfigurationCache"   | true
+        "supportsConfigurationCache" | false
```

Comment:
Rather than using `@Unroll` like this I'd just have one test for each. The boilerplate for the test is very minimal, so not much duplication. Two separate tests are easier to understand, and you cannot rerun a single `@Unroll`ed test in Intellij.